### PR TITLE
Add fill arg to Polygon

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Add `fill` arg to `Polygon` function to realize the [content of the document](https://docs.pyvista.org/user-guide/vtk_to_pyvista.html#tradeoffs).

```python
import pyvista


geom1 = pyvista.Polygon(fill=True)
geom2 = pyvista.Polygon(fill=False)

pl = pyvista.Plotter(shape=(1, 2))
pl.subplot(0, 0)
pl.add_text("fill=True")
pl.add_mesh(geom1)
pl.subplot(0, 1)
pl.add_text("fill=False")
pl.add_mesh(geom2)
pl.show(cpos="xy")
```
test

 

### Details

- Inspired from https://docs.pyvista.org/user-guide/vtk_to_pyvista.html#tradeoffs
- [VTK document](https://vtk.org/doc/nightly/html/classvtkRegularPolygonSource.html#a35a78b78294e362cfd5c46494e048561).

### CODE REVIEW

File: tests/test_geometric_objects.py [hunk 1]
```diff
@@ -141,6 +141,11 @@ def test_polygon():
     geom = pyvista.Polygon()
     assert np.any(geom.points)
 
+    geom1 = pyvista.Polygon(generate_polygon=True)
+    assert geom.n_cells == 2
```
Comment (tkoyama010) [inline tests/test_geometric_objects.py:145]:
```suggestion
    assert geom1.n_cells == 2
```

File: tests/test_geometric_objects.py [hunk 2]
```diff
@@ -141,6 +141,11 @@ def test_polygon():
...
+    geom2 = pyvista.Polygon(generate_polygon=False)
+    assert geom.n_cells == 1
```
Comment (tkoyama010) [inline tests/test_geometric_objects.py:147]:
```suggestion
    assert geom2.n_cells == 1
```

File: pyvista/utilities/geometric_objects.py [hunk 3]
```diff
@@ -768,6 +768,9 @@ def Polygon(center=(0.0, 0.0, 0.0), radius=1, normal=(0, 0, 1), n_sides=6):
     n_sides : int, optional
         Number of sides of the polygon.
 
+    generate_polygon : bool, optional
```
Comment (adeak) [inline pyvista/utilities/geometric_objects.py:771]: If possible we should give a more meaningful name to this. Something like `fill` or similar. Anything "polygon" seems confusing in a callable called "Polygon". A better name would also make it clear which value of the kwarg leads to filling.

File: pyvista/utilities/geometric_objects.py [hunk 4]
```diff
@@ -783,6 +786,10 @@ def Polygon(center=(0.0, 0.0, 0.0), radius=1, normal=(0, 0, 1), n_sides=6):
 
     """
     src = _vtk.vtkRegularPolygonSource()
+    if generate_polygon:
+        src.GeneratePolygonOn()
+    else:
+        src.GeneratePolygonOff()
```
Comment (adeak) [inline pyvista/utilities/geometric_objects.py:792]: Knowing VTK there's probably a method we can call as `src.SetGeneratePolygon(generate_polygon)`.

File: pyvista/utilities/geometric_objects.py [hunk 5]
```diff
@@ -768,6 +768,9 @@ def Polygon(center=(0.0, 0.0, 0.0), radius=1, normal=(0, 0, 1), n_sides=6):
     n_sides : int, optional
         Number of sides of the polygon.
 
+    generate_polygon : bool, optional
+        Enable or disable the producing filled polygons.
```
Comment (adeak) [inline pyvista/utilities/geometric_objects.py:772]:
```suggestion
        Enable or disable producing filled polygons.
```

File: pyvista/utilities/geometric_objects.py (hunk 4 above)
Comment (tkoyama010) [inline pyvista/utilities/geometric_objects.py:792]:
```suggestion
    src.SetGeneratePolygon(generate_polygon)
```

Comment (tkoyama010) [reply pyvista/utilities/geometric_objects.py:792]: It is worth [trying](https://github.com/pyvista/pyvista/pull/2767/files#r894998106).

File: pyvista/utilities/geometric_objects.py (hunk 3 above)
Comment (tkoyama010) [reply pyvista/utilities/geometric_objects.py:771]: I didn't notice `fill` naming. Thanks.
```suggestion
    fill : bool, optional
```

File: pyvista/utilities/geometric_objects.py [hunk 6]
```diff
@@ -750,7 +750,7 @@ def Cone(
     return pyvista.wrap(src.GetOutput())
 
 
-def Polygon(center=(0.0, 0.0, 0.0), radius=1, normal=(0, 0, 1), n_sides=6):
+def Polygon(center=(0.0, 0.0, 0.0), radius=1, normal=(0, 0, 1), n_sides=6, generate_polygon=True):
```
Comment (tkoyama010) [inline pyvista/utilities/geometric_objects.py:753]:
```suggestion
def Polygon(center=(0.0, 0.0, 0.0), radius=1, normal=(0, 0, 1), n_sides=6, fill=True):
```

File: tests/test_geometric_objects.py [hunk 7]
```diff
@@ -141,6 +141,11 @@ def test_polygon():
     geom = pyvista.Polygon()
     assert np.any(geom.points)
 
+    geom1 = pyvista.Polygon(generate_polygon=True)
+    assert geom1.n_cells == 2
+    geom2 = pyvista.Polygon(generate_polygon=False)
+    assert geom2.n_cells == 1
+
```
Comment (tkoyama010) [inline tests/test_geometric_objects.py:148]:
```suggestion
    geom1 = pyvista.Polygon(fill=True)
    assert geom1.n_cells == 2
    geom2 = pyvista.Polygon(fill=False)
    assert geom2.n_cells == 1

```

File: pyvista/utilities/geometric_objects.py [hunk 8]
```diff
@@ -783,6 +786,7 @@ def Polygon(center=(0.0, 0.0, 0.0), radius=1, normal=(0, 0, 1), n_sides=6):
 
     """
     src = _vtk.vtkRegularPolygonSource()
+    src.SetGeneratePolygon(generate_polygon)
```
Comment (tkoyama010) [inline pyvista/utilities/geometric_objects.py:789]:
```suggestion
    src.SetGeneratePolygon(generate_polygon)
```
```suggestion
    src.SetGeneratePolygon(fill)
```

**Review by adeak — APPROVED** [review]
LGTM, thanks!

**Comment by tkoyama010** [comment]
Suggest to merge at 2022/6/12

**Pull request merged**